src/WaterFlow.jl

```diff
@@ -0,0 +1,93 @@
+# These are routes to perform waterflow calculations (upstream area etc.) on a DEM
+
+using WhereTheWaterFlows
+import WhereTheWaterFlows: waterflows
+
+export waterflows
+
+"""
+    dlon, dlat = spacing(lon,lat)
+
+Computes the spacing with central differences
+"""
+function spacing(lon,lat)
+    dlon = zero(lon.val[:,:,1])
+    dlat = zero(lat.val[:,:,1])
```

Comment:
```suggestion
    dlon = zero(size(lon.val)[1:2])
    dlat = zero(size(lat.val)[1:2])
```
This will avoid allocating two intermediate arrays